web/template/add-firm.gotmpl

```diff
@@ -173,7 +173,8 @@
                     </button>
                     <a
                         class="govuk-link"
-                        href="{{ prefix (printf "/deputy/%d/change-firm" .DeputyId) }}">
+                        href="javascript: history.go(-1)"
```

Comment:
Did you mean to make this change?

Reply:
Yes. Phil told me he'd found a bug where if you went to the change firm page and decided to add a firm and then pressed cancel you were trapped on the change-firm page as the javascript back button wouldn't take you back to the deputy details page (because the last page you were on was add a firm). 
If you're not happy with this I should be able to do the same fix by putting a cancel in the change-firm page which takes you back to deputy details

Reply:
Not sure I understand the issue. The cancel button as it was had defined behaviour - it always would navigate back to the Change Firm page. Is that a bug?

Reply:
There is actually no cancel button on the prototype either 
![Uploading Screenshot 2021-12-06 at 11.19.02.png…]()

Reply:
@denialanderror the page structure means that you can get trapped on the change-firm page. 
the change-firm page looks as above. If you click 'add new firm' it takes you to a new page. If you click 'cancel' on add new firm page it takes you to change-firm page but you are trapped there. Pressing back will take you back to 'add new firm' page and there is no cancel button (including on the prototype). 
My solution was to use the javascript but I can raise it to Jen if you think it's a design issue instead.